presto-hive/src/main/java/io/prestosql/plugin/hive/util/ConfigurationUtils.java

```diff
@@ -64,7 +64,24 @@ public static JobConf toJobConf(Configuration conf)
         if (conf instanceof JobConf) {
             return (JobConf) conf;
         }
-        return new JobConf(conf);
+        return new JobConf(conf)
+        {
+            @Override
+            public Class<?> getClassByNameOrNull(String name)
+            {
+                Class<?> clazz = super.getClassByNameOrNull(name);
+                if (name != null && clazz == null) {
+                    // fallback
+                    switch (name) {
+                        case "com.hadoop.compression.lzo.LzoCodec":
```

Comment:
Should we just provide those classes in `trino-hive-apache` as shims that delegate to Airlift implementations instead?

Reply:
That would cause classpath conflicts for anyone adding the compression codecs to the plugin classpath.

@electrum do you have an opinion?

Reply:
> conflicts for anyone adding the compression codecs to the plugin classpath.

We usually do not support hacking, don't we? 

> we would need to have some transition period,

Does it mean we encourage users to place jars manually somewhere?

Reply:
> Does it mean we encourage users to place jars manually somewhere?

i do not. but was trying to be empathetic if they do

Reply:
I agree, while this is clever, shims would be cleaner. I’m not concerned with breaking anyone who added additional JARs. This will be better for them anyway as they won’t need to do it anymore.